force-app/main/default/classes/AffiliationMappingsService.cls

```diff
@@ -101,6 +101,15 @@ public virtual with sharing class AffiliationMappingsService {
         this.locateAffiliationMappingsMapper().updateAffiliationMappings(afflMappingModel);
     }
 
+    /*****************************************************************************
+     * @description Checks user permissions and passes AffiliationMappingsModel to
+     * updateAffiliationMappings.
+     *****************************************************************************/
+    public void updateAutoEnrolllmentMapping(AffiliationMappingsModel affiliationMappingsModel) {
```

Comment:
You may need a slightly different entry method as an "update" may span two mappings.

For instance, if I update the Academic Program Mapping to instead be the record type for Business Organization, I'll need to determine the difference in those and update both records accordingly. The expected behavior for this would be to that Academic Program would have its auto enrollment, status, and role fields cleared while the Business Organization mapping would have its fields set to the new values.

The inability to select an unmapped account record type should protect us from almost all collision issues, at least until we separate the mappings from a data standpoint.

Reply:
Making sure that i understand it correctly:

1) For auto enrollment editing, we are only updating status and role for 'enabled' auto enrollment and we don't allow users to update account record types.

2) Updating account record types should fall on affiliation mappings settings page and we might need to refactor the updating affiliations method so that we can update both affiliation mappings.

Either way I think we need to update the user stories too.

Reply:
Not quite on that. More that "editing" the Account Record type should work seamlessly from the user's standpoint, but what we need to do is to have it update 2 back end records instead of 1.